src/marked.js

```diff
@@ -164,6 +154,50 @@ marked.use = function(extension) {
   marked.setOptions(opts);
 };
 
+/**
+ * Iterate over every token
```

Comment:
Possibly worth documenting a few things here:

1. That this is recursive (could be a source of weird bugs on huge documents).
2. That the visit order is a pre-order DFS.
3. That if callback returns false, the iteration ceases.